app/src/main/java/co/nimblehq/ui/base/BaseActivity.kt

```diff
@@ -1,19 +1,51 @@
 package co.nimblehq.ui.base
 
+import android.os.Bundle
+import androidx.annotation.LayoutRes
 import androidx.appcompat.app.AppCompatActivity
+import co.nimblehq.extension.userReadableMessage
+import co.nimblehq.ui.common.Toaster
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.addTo
+import javax.inject.Inject
 
-abstract class BaseActivity: AppCompatActivity() {
+abstract class BaseActivity : AppCompatActivity() {
 
-    private var disposables = CompositeDisposable()
+    @Inject
+    lateinit var toaster: Toaster
 
-    protected fun Disposable.bindForDisposable() {
-        disposables.add(this)
+    @get:LayoutRes
+    protected abstract val layoutRes: Int
+
+    private val disposables = CompositeDisposable()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(layoutRes)
+
+        bindViewModel()
     }
 
     override fun onDestroy() {
         super.onDestroy()
-        disposables.clear()
+        disposables.dispose()
+    }
+
+    abstract fun bindViewModel()
+
+    protected fun Disposable.addToDisposables() = addTo(disposables)
+
+    protected inline infix fun <T> Observable<T>.bindTo(crossinline action: (T) -> Unit) {
+        observeOn(AndroidSchedulers.mainThread())
```

Comment:
Should it be `SchedulerProvider().main()` 🙏  same with BaseFragment

Reply:
I think this is unnecessary to initialize `SchedulerProvider` whenever we use `bindTo`. So use static method from `AndroidSchedulers` is still good imo 🙏

Reply:
What about if we inject 🤔 If not work so let's use the same one 🙏 

```
 @Inject
 lateinit var schedulerProvider: SchedulerProvider

 observeOn(schedulerProvider.main())
            .subscribe { action(it) }
            .addToDisposables()
```

Reply:
uhm yeah, it works, updated 🙏